sdk/core/core-http/lib/webResource.ts

```diff
@@ -147,35 +167,55 @@ export class WebResource {
+        throw new Error(
+          'The provided method "' +
+            options.method +
+            '" is invalid. Supported HTTP methods are: ' +
+            JSON.stringify(validMethods)
+        );
       }
     }
-    this.method = (options.method.toUpperCase() as HttpMethods);
+    this.method = options.method.toUpperCase() as HttpMethods;
 
     // construct the url if path template is provided
     if (options.pathTemplate) {
       const { pathTemplate, pathParameters } = options;
       if (typeof pathTemplate !== "string") {
-        throw new Error("options.pathTemplate must be of type \"string\".");
+        throw new Error('options.pathTemplate must be of type "string".');
       }
       if (!options.baseUrl) {
         options.baseUrl = "https://management.azure.com";
       }
       const baseUrl = options.baseUrl;
-      let url = baseUrl + (baseUrl.endsWith("/") ? "" : "/") + (pathTemplate.startsWith("/") ? pathTemplate.slice(1) : pathTemplate);
-      const segments = url.match(/({\w*\s*\w*})/ig);
+      let url =
+        baseUrl +
+        (baseUrl.endsWith("/") ? "" : "/") +
+        (pathTemplate.startsWith("/") ? pathTemplate.slice(1) : pathTemplate);
+      const segments = url.match(/({\w*\s*\w*})/gi);
       if (segments && segments.length) {
         if (!pathParameters) {
-          throw new Error(`pathTemplate: ${pathTemplate} has been provided. Hence, options.pathParameters must also be provided.`);
+          throw new Error(
+            `pathTemplate: ${pathTemplate} has been provided. Hence, options.pathParameters must also be provided.`
+          );
         }
-        segments.forEach(function (item) {
+        segments.forEach(function(item) {
           const pathParamName = item.slice(1, -1);
           const pathParam = (pathParameters as { [key: string]: any })[pathParamName];
-          if (pathParam === null || pathParam === undefined || !(typeof pathParam === "string" || typeof pathParam === "object")) {
-            throw new Error(`pathTemplate: ${pathTemplate} contains the path parameter ${pathParamName}` +
-              ` however, it is not present in ${pathParameters} - ${JSON.stringify(pathParameters, undefined, 2)}.` +
-              `The value of the path parameter can either be a "string" of the form { ${pathParamName}: "some sample value" } or ` +
-              `it can be an "object" of the form { "${pathParamName}": { value: "some sample value", skipUrlEncoding: true } }.`);
+          if (
+            pathParam === null ||
+            pathParam === undefined ||
+            !(typeof pathParam === "string" || typeof pathParam === "object")
+          ) {
+            throw new Error(
+              `pathTemplate: ${pathTemplate} contains the path parameter ${pathParamName}` +
+                ` however, it is not present in ${pathParameters} - ${JSON.stringify(
+                  pathParameters,
+                  undefined,
+                  2
+                )}.` +
```

Comment:
This one is very hard to read. Please pull `JSON.stringify(pathParameters, undefined, 2)` out into a constant instead of using it inline. This way the formatter should treat it more kindly